### PR TITLE
[CIS-1281] Improve `synchronize` messaging

### PR DESCRIPTION
**Pull request by adamrushy**
### 🔗 Issue Link
CIS-1281

### 🎯 Goal

The improvements for `synchronize`. This PR is phase 1 of this project to really start improving how customers use `synchronize` and more importantly _when_ to use it. This PR is mainly focussing our documentation and guides as we move across to a more sustainable architecture in the near future.

### 🛠 Implementation

Mainly documentation, I have implemented a whole new guide focussed solely on `synchronize()` and added these tips throughout other documentation pieces. I have also added inline docs for the use of `setUp()`.

### 🧪 Testing

The testing here is mainly in the docs, have I covered everything?

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [x] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

**Review by bielikb — DISMISSED** [review]

File: docusaurus/docs/iOS/basics/getting-started.md [hunk 1]
```diff
@@ -86,3 +86,35 @@ When deciding to push your `UIViewController` on to the `NavigationStack`, you c
...
 
 `ChannelListQuery` allows us to define the channels to fetch and their order. Here we are listing channels where the current user is a member. By default tapping on a channel will navigate to `ChatMessageListVC`.
+
+### Creating a Channel
+
+You now have your very first Stream Chat app showing a list of Channels, but you're probably wondering how you can create your very first channel.
+
+```swift
+do {
+    try controller(createChannelWithId: ChannelId(type: .livestream, id: UUID().uuidString), name: channelName)
```
Comment (b-onc) [inline docusaurus/docs/iOS/basics/getting-started.md:96]: Is there a typo here?
```suggestion
    try client.channelController(createChannelWithId: ChannelId(type: .livestream, id: UUID().uuidString), name: channelName)
```

File: docusaurus/docs/iOS/basics/getting-started.md [hunk 2]
```diff
@@ -86,3 +86,35 @@ When deciding to push your `UIViewController` on to the `NavigationStack`, you c
...
+
+You can access `createChannelWithId:` function on the `ChannelController` which allows you to pass some parameters and create your very first channel.
+
+The channel `type` is an enum that describes what the channel's intention is.
+
+Your `ChannelId` has to be a unique ID and you can set this to anything, in this example we're using the `UUID()` provided by Apple. Finally, you can pass through the name of the channel which is a `String` and also some additional parameters if required.
+
+:::tip Using Synchronize
+
+After creating the channel `try controller(createChannelWithId: ChannelId(type: .livestream, id: UUID().uuidString), name: channelName)` it's important you call `synchronize()` after so the local and remote data is updated. You can read more about the importance of `synchronize()` [here](../../guides/importance-of-synchronize)..
```
Comment (b-onc) [inline docusaurus/docs/iOS/basics/getting-started.md:116]: same as above?

File: docusaurus/docs/iOS/guides/importance-of-synchronize.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+In addition, you don't need to call `synchronize()` to be able to call actions on the channel, such as `freezeChannel`:
+
+```swift
+let channelController = chatClient.channelController(for: ChannelId(type: .messaging, id: "general"))
+channelController.freezeChannel()
+```
+
+But as stated in the table above, if the Channel is not created in the backend yet, you'll need to call `synchronize()` first, else the action calls will fail.
+
+`Synchronize()` is automatically called within the `setUp()` function on each `View` or `ViewController` which will automatically update both local and remote data for you. It's important to note that if you implement your own custom implementation that you call `super.setUp()` so this function is called. If you implement your own UI completely then make sure you call `synchronize()` on the custom UI lifecycle.
```
Comment (b-onc) [inline docusaurus/docs/iOS/guides/importance-of-synchronize.md:28]:
```suggestion
`synchronize()` is automatically called within the `setUp()` function on each `View` or `ViewController` which will automatically update both local and remote data for you. It's important to note that if you implement your own custom implementation that you call `super.setUp()` so this function is called. If you implement your own UI completely then make sure you call `synchronize()` on the custom UI lifecycle.
```

File: docusaurus/docs/iOS/guides/importance-of-synchronize.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+If you only need the local data, you can just access it after creation, like so:
+
+```swift
+let channelController = chatClient.channelController(for: ChannelId(type: .messaging, id: "general"))
+let channel = channelController.channel
+let messages = channelController.messages
+ ```
+
+In addition, you don't need to call `synchronize()` to be able to call actions on the channel, such as `freezeChannel`:
```
Comment (b-onc) [inline docusaurus/docs/iOS/guides/importance-of-synchronize.md:19]: This is only true for channels with id's passed. If one uses `createChannelWithMembers:`, they need to call `synchronize` to be able to call actions on the channel

File: docusaurus/docs/iOS/basics/getting-started.md (hunk 1 above)
Comment (adamrushy) [reply docusaurus/docs/iOS/basics/getting-started.md:96]: Good spot 👍🏼

File: Sources/StreamChatUI/CommonViews/BaseViews.swift [hunk 5]
```diff
@@ -344,6 +344,11 @@ open class _ViewController: UIViewController, Customizable {
         updateContent()
     }
     
+    /// `setUp()` is an important function within the ViewController lifecycle
+    /// It's resposibility is to set the delegates and also call `synchronize()`
+    /// this will make sure your local & remote data is in sync.
+    /// If you override this method without calling `super.setUp()` it's important
+    /// to make sure `synchronize()` is called.
```
Comment (bielikb) [inline Sources/StreamChatUI/CommonViews/BaseViews.swift:351]: @adamrushy to be more explicit about the intent here, this paragraph could probably leverage the [important markup callout](https://developer.apple.com/library/archive/documentation/Xcode/Reference/xcode_markup_formatting_ref/Important.html#//apple_ref/doc/uid/TP40016497-CH37-SW1).

Comment (adamrushy) [reply Sources/StreamChatUI/CommonViews/BaseViews.swift]: Nice idea 👍🏼

**Review by bielikb — DISMISSED** [review]

**Review by bielikb — APPROVED** [review]

**Pull request merged**